src/components/traces/traceDetails.jsx

```diff
@@ -16,19 +16,61 @@
  */
 
 import React from 'react';
+import PropTypes from 'prop-types';
+import { observer } from 'mobx-react';
 
 import './traceDetails.less';
-import WorkInProgress from '../common/workInProgress';
-
-export default () => (
-    <section className="trace-details">
-        <div className="trace-details-nav">
-            <h4>Timeline</h4>
-            <div className="trace-details-toolbar btn-group">
-                <a className="btn btn-default"><span className="trace-details-toolbar-option-icon ti-share"/> Raw Trace</a>
-                <a className="btn btn-primary"><span className="trace-details-toolbar-option-icon ti-link"/> Copy Link</a>
-            </div>
-        </div>
-        <WorkInProgress/>
-    </section>
-);
+import TraceDetailsRow from './traceDetailsRow';
+import Loading from '../common/loading';
+import Error from '../common/error';
+import activeTraceStore from '../../stores/activeTraceStore';
+
+
+@observer
+export default class TraceDetails extends React.Component {
+    static propTypes = {
+        traceId: PropTypes.string.isRequired
+    };
+
+    componentDidMount() {
+        activeTraceStore.fetchTraceDetails(this.props.traceId);
+    }
+
+    render() {
+        const rowChildren = activeTraceStore.spans.map((span, index) =>
+            (<TraceDetailsRow
+                key={span.id}
+                index={index}
+                startTime={activeTraceStore.startTime}
+                rowHeight={22}
+                rowPadding={10}
+                span={span}
+                totalDuration={activeTraceStore.totalDuration}
+            />));
+
+        const svgHeight = {height: 30 + (42 * activeTraceStore.spans.length)};
+
+        return (
+            <section className="trace-details">
+                <div className="trace-details-nav">
+                    <h4>Trace Timeline</h4>
+                    <div className="trace-details-toolbar btn-group">
+                        <a className="btn btn-default"><span className="trace-details-toolbar-option-icon ti-share"/> Raw Trace</a>
+                        <a className="btn btn-primary"><span className="trace-details-toolbar-option-icon ti-link"/> Copy Link</a>
+                    </div>
+                </div>
+                <div>
+                    <h4>ID: {this.props.traceId}</h4>
+                    { activeTraceStore.promiseState && activeTraceStore.promiseState.case({
+                        pending: () => <Loading />,
+                        rejected: () => <Error />,
+                        fulfilled: () => ((activeTraceStore.spans && activeTraceStore.spans.length)
+                            ? (<svg style={svgHeight} className="trace-details-graph">{rowChildren}</svg>)
```

Comment:
Should add a viewBox attribute in svg to make sure we have standard definition of a pixel, ref - https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Positions